compute/oslogin/new_service_account_ssh_test.py

```diff
@@ -0,0 +1,204 @@
+    firewall_client.insert(request).result()
+
+    yield firewall_client.get(project=PROJECT, firewall=TEST_ID)
+    try:
+        firewall_client.delete(project=PROJECT, firewall=TEST_ID)
+    except NotFound:
+        # That means the GCE Enforcer deleted it before us
+        pass
+
+
+@pytest.fixture()
+def oslogin_instance(ssh_firewall, oslogin_service_account):
+    instance = compute_v1.Instance()
+    instance.name = TEST_ID
+
+    disk = compute_v1.AttachedDisk()
+    initialize_params = compute_v1.AttachedDiskInitializeParams()
+    initialize_params.source_image = (
+        "projects/ubuntu-os-cloud/global/images/family/ubuntu-2204-lts"
+    )
+    initialize_params.disk_size_gb = 25
+    initialize_params.disk_type = f"zones/{ZONE}/diskTypes/pd-standard"
+    disk.initialize_params = initialize_params
+    disk.auto_delete = True
+    disk.boot = True
+
+    network_interface = compute_v1.NetworkInterface()
+    network_interface.name = "global/networks/default"
+
+    access = compute_v1.AccessConfig()
+    access.type_ = compute_v1.AccessConfig.Type.ONE_TO_ONE_NAT.name
+    access.name = "External NAT"
+    access.network_tier = access.NetworkTier.PREMIUM.name
+    network_interface.access_configs = [access]
+
+    instance.disks = [disk]
+    instance.machine_type = f"zones/{ZONE}/machineTypes/e2-micro"
+    instance.network_interfaces = [network_interface]
+    instance.tags = compute_v1.Tags()
+    instance.tags.items = [FIREWALL_TAG]
+
+    sa = compute_v1.ServiceAccount()
+    sa.email = oslogin_service_account['name']
+    sa.scopes = ['https://www.googleapis.com/auth/cloud-platform']
+    instance.service_accounts = [sa]
+
+    instance.metadata = compute_v1.Metadata()
+    item = compute_v1.Items()
+    item.key = 'enable-oslogin'
+    item.value = 'TRUE'
+    instance.metadata.items = [item]
+    instance.zone = ZONE
+
+    client = compute_v1.InstancesClient()
+    request = compute_v1.InsertInstanceRequest()
+    request.project = PROJECT
+    request.instance_resource = instance
+    request.zone = ZONE
+
+    client.insert(request).result()
+
+    policy = compute_v1.ZoneSetPolicyRequest()
+    binding = compute_v1.Binding()
+    binding.members = [f'serviceAccount:{oslogin_service_account["name"]}']
+    binding.role = 'roles/compute.osLogin'
+    policy.bindings = [binding]
+    client.set_iam_policy(project=PROJECT, zone=ZONE, resource=TEST_ID, zone_set_policy_request_resource=policy)
+
+    # Wait for everything to propagate
+    time.sleep(5)
+
+    yield client.get(project=PROJECT, zone=ZONE, instance=instance.name)
+
+    client.delete(project=PROJECT, zone=ZONE, instance=instance.name).result()
+
+
+def test_oslogin_ssh(oslogin_instance, oslogin_service_account, capsys):
+    account = 'users/' + oslogin_service_account['name']
+    oslogin_client = oslogin_v1.OsLoginServiceClient(credentials=oslogin_service_account['credentials'])
+    # Letting everything settle down...
+    time.sleep(60)
```

Comment:
Should we use a try-except afterwards in case this time wasn't long enough?

Reply:
If this time was not long enough, then I think we have a problem with the resource setup. The code already waits for completion of operations like instance creation. IMO, 60 seconds is quite a safe margin.

Reply:
SGTM.